src/main/java/com/Alvaeron/api/RPEngineAPI.java

```diff
@@ -45,83 +47,111 @@ public static RoleplayPlayer getRoleplayPlayer(UUID uuid) {
      */
     @SuppressWarnings("deprecation")
     public static RoleplayPlayer getRoleplayPlayer(String playerName) {
-        if (Bukkit.getPlayer(playerName) != null) {
-            return Engine.manager.getPlayer(Bukkit.getPlayer(playerName).getUniqueId());
-        } else if (Bukkit.getOfflinePlayer(playerName) != null) {
-            Player p = (Player) Bukkit.getOfflinePlayer(playerName);
-            if (Engine.manager.getPlayer(p.getUniqueId()) != null) {
-                return Engine.manager.getPlayer(p.getUniqueId());
-            } else {
-                Engine.mm.createRoleplayPlayer(p);
-                return Engine.manager.getPlayer(p.getUniqueId());
-            }
-        } else {
+        Player player = Bukkit.getPlayer(playerName);
+        if (player != null) {
+            return Engine.manager.getPlayer(player.getUniqueId());
+        }
+
+        OfflinePlayer p = Bukkit.getOfflinePlayer(playerName);
+        if (p == null) {
```

Comment:
`getOfflinePlayer()` cannot return null. See: https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/Bukkit.html#getOfflinePlayer-java.util.UUID-

Consider changing to `if (!p.hasPlayedBefore())`

Reply:
I saw that but I still don't trust it, especially since this particular method is deprecated. I'll add that check in addition =)

Reply:
`getOfflinePlayer(UUID)` is not deprecated. `getOfflinePlayer(String)` is

Reply:
We will want to move away from player names in a separate commit